lib/xnvme_be_posix_async_thrpool.c

```diff
@@ -203,10 +203,12 @@ _posix_async_thrpool_term(struct xnvme_queue *q)
 		XNVME_DEBUG("FAILED: pthread_mutex_unlock(), err_lock: %d", err_lock);
 	}
 
-	for (int i = 0; i < queue->nthreads; i++) {
-		pthread_join(queue->threads[i], NULL);
+	if (queue->threads) {
+		for (int i = 0; i < queue->nthreads; i++) {
```

Comment:
As i recall, then ``queue->nthreads`` is ``0`` in case of error.
But I guess scan-build does not see that relation.

How about for "compactness" changing this to the following instead:
```
for (int i = 0; queue->threads && (i < queue->nthreads); i++) {
   thread_join(queue->threads[i], NULL);
}
```

Reply:
Done